gcloud-java-dns/src/test/java/com/google/gcloud/dns/ChangeRequestTest.java

```diff
@@ -16,203 +16,132 @@
-    assertEquals(partial, partial.toBuilder().build());
-    partial = ChangeRequest.builder().startTimeMillis(START_TIME_MILLIS).build();
-    assertEquals(partial, partial.toBuilder().build());
-    partial = ChangeRequest.builder().status(STATUS).build();
-    assertEquals(partial, partial.toBuilder().build());
-  }
-
-  @Test
-  public void testClearAdditions() {
-    ChangeRequest clone = CHANGE.toBuilder().clearAdditions().build();
-    assertTrue(clone.additions().isEmpty());
-    assertFalse(clone.deletions().isEmpty());
+  private static final DnsOptions OPTIONS = createStrictMock(DnsOptions.class);
+
+  private Dns dns;
+  private ChangeRequest changeRequest;
+  private ChangeRequest changeRequestPartial;
+
+  @Before
+  public void setUp() throws Exception {
+    dns = createStrictMock(Dns.class);
+    expect(dns.options()).andReturn(OPTIONS);
+    expect(dns.options()).andReturn(OPTIONS);
+    replay(dns);
+    changeRequest = new ChangeRequest(dns, ZONE_NAME, new ChangeRequestInfo.BuilderImpl(
+        CHANGE_REQUEST_INFO.toBuilder()
+            .startTimeMillis(132L)
+            .id("12")
+            .status(ChangeRequest.Status.DONE)
+            .build()));
+    changeRequestPartial = new ChangeRequest(dns, ZONE_NAME,
+        new ChangeRequest.BuilderImpl(CHANGE_REQUEST_INFO));
+    reset(dns);
   }
 
-  @Test
-  public void testAddAddition() {
-    try {
-      CHANGE.toBuilder().add(null);
-      fail("Should not be able to add null RecordSet.");
-    } catch (NullPointerException e) {
-      // expected
-    }
-    ChangeRequest clone = CHANGE.toBuilder().add(RECORD1).build();
-    assertEquals(CHANGE.additions().size() + 1, clone.additions().size());
+  @After
+  public void tearDown() throws Exception {
+    verify(dns);
   }
 
   @Test
-  public void testAddDeletion() {
-    try {
-      CHANGE.toBuilder().delete(null);
-      fail("Should not be able to delete null RecordSet.");
-    } catch (NullPointerException e) {
-      // expected
-    }
-    ChangeRequest clone = CHANGE.toBuilder().delete(RECORD1).build();
-    assertEquals(CHANGE.deletions().size() + 1, clone.deletions().size());
+  public void testConstructor() {
+    replay(dns);
+    assertEquals(CHANGE_REQUEST_INFO.toPb(), changeRequestPartial.toPb());
+    assertNotNull(changeRequest.dns());
+    assertEquals(ZONE_NAME, changeRequest.zoneName());
+    assertNotNull(changeRequestPartial.dns());
+    assertEquals(ZONE_NAME, changeRequestPartial.zoneName());
   }
 
   @Test
-  public void testClearDeletions() {
-    ChangeRequest clone = CHANGE.toBuilder().clearDeletions().build();
-    assertTrue(clone.deletions().isEmpty());
-    assertFalse(clone.additions().isEmpty());
+  public void testFromPb() {
+    expect(dns.options()).andReturn(OPTIONS);
+    expect(dns.options()).andReturn(OPTIONS);
+    replay(dns);
+    assertEquals(ChangeRequest.fromPb(dns, ZONE_NAME, changeRequest.toPb()), changeRequest);
```

Comment:
Nit: first parameter should be the expected value, second parameter should be the value under test (also in the following assertion)